server/plugin/command_test.go

```diff
@@ -83,3 +83,72 @@ func TestValidateFeatures(t *testing.T) {
 		})
 	}
 }
+
+func TestParseCommand(t *testing.T) {
+	type output struct {
+		command    string
+		action     string
+		parameters []string
+	}
+
+	tt := []struct {
+		name  string
+		input string
+		want  output
+	}{
+		{
+			name:  "no parameters",
+			input: "/github subscribe",
+			want: output{
+				"/github",
+				"subscribe",
+				[]string(nil),
+			},
+		},
+		{
+			name:  "no action and no parameters",
+			input: "/github",
+			want: output{
+				"/github",
+				"",
+				[]string(nil),
+			},
+		},
+		{
+			name:  "simple one-word label",
+			input: "/github subscribe DHaussermann/hello-world issues,label:\"Help\"",
+			want: output{
+				"/github",
+				"subscribe",
+				[]string{"DHaussermann/hello-world", "issues,label:\"Help\""},
+			},
+		},
+		{
+			name:  "two-word label",
+			input: "/github subscribe DHaussermann/hello-world issues,label:\"Help Wanted\"",
+			want: output{
+				"/github",
+				"subscribe",
+				[]string{"DHaussermann/hello-world", "issues,label:\"Help Wanted\""},
+			},
+		},
+		{
+			name:  "multi-word label",
+			input: "/github subscribe DHaussermann/hello-world issues,label:\"Good First Issue\"",
+			want: output{
+				"/github",
+				"subscribe",
+				[]string{"DHaussermann/hello-world", "issues,label:\"Good First Issue\""},
+			},
+		},
```

Comment:
I know they are corner cases, but I would like to see in the tests:
- Weird spacing (words separated by more than one space, both in between quotes and outside the quotes)
- Trailing spaces (spaces before and after the command)
- Line breaks

Reply:
@larkox I added test cases for multiple spaces inside & outside double quotes, but I'm not sure if it's really necessary to accommodate for line breaks. When the user hits "enter" while typing, the command will be submitted anyway, right?

Reply:
I can type shift+enter to add a new line. It is an edge case that will not happen often but I think we should make sure we are handling it.

Reply:
@larkox Added the test case: https://github.com/mattermost/mattermost-plugin-github/pull/336/commits/c98c8fdc72cc55d92c6c2c154c5176a56eec1c44 
Apparently my new implementation already handles line breaks :smile: